Composer/packages/webchat-client/src/utils/ConversationService.ts

```diff
@@ -0,0 +1,316 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+import { v4 as uuidv4 } from 'uuid';
+import axios, { AxiosInstance } from 'axios';
+import { createDirectLine } from 'botframework-webchat';
+
+export const headers = {
+  'Content-Accept': 'application/json',
+};
+
+export type User = {
+  id: string;
+  name: string;
+  role: string;
+};
+
+export type BotSecrets = { msAppId: string; msPassword: string };
+export type ChannelService = 'public' | 'usgov';
+
+type WebChatMode = 'livechat' | 'transcript';
+type ConversationMember = {
+  id: string;
+  name: string;
+  role?: string;
+};
+
+type StartConversationPayload = {
+  bot?: ConversationMember;
+  botUrl: string;
+  channelServiceType: ChannelService;
+  members: ConversationMember[];
+  mode: WebChatMode;
+  locale: string;
+  msaAppId?: string;
+  msaPassword?: string;
+};
+
+// All activity types. Just parsing for Trace currently.
+export enum ActivityTypes {
+  Message = 'message',
+  ContactRelationUpdate = 'contactRelationUpdate',
+  ConversationUpdate = 'conversationUpdate',
+  Typing = 'typing',
+  EndOfConversation = 'endOfConversation',
+  Event = 'event',
+  Invoke = 'invoke',
+  InvokeResponse = 'invokeResponse',
+  DeleteUserData = 'deleteUserData',
+  MessageUpdate = 'messageUpdate',
+  MessageDelete = 'messageDelete',
+  InstallationUpdate = 'installationUpdate',
+  MessageReaction = 'messageReaction',
+  Suggestion = 'suggestion',
+  Trace = 'trace',
+  Handoff = 'handoff',
+}
+
+export type ChatData = {
+  chatMode: 'livechat' | 'transcript';
+  directline: any;
+  projectId: string;
+  user: User;
+  conversationId: string;
+};
+
+export default class ConversationService {
```

Comment:
It's better if the methods of the class are prototypal (method(args){...}) vs. arrow (method: (args) => {...})
I highly recommend converting your code to use prototype.